homeassistant/components/sensor/radarr.py

```diff
@@ -0,0 +1,230 @@
+                    self.urlbase, start, end),
+                headers={'X-Api-Key': self.apikey},
+                timeout=5)
+        except OSError:
+            _LOGGER.error('Host %s is not available', self.host)
+            self._available = False
+            self._state = None
+            return
+
+        if res.status_code == 200:
+            if self.type in ['upcoming', 'movies', 'commands']:
+                self.data = res.json()
+                self._state = len(self.data)
+            elif self.type == 'diskspace':
+                # If included paths are not provided, use all data
+                if self.included == []:
+                    self.data = res.json()
+                else:
+                    # Filter to only show lists that are included
+                    self.data = list(
+                        filter(
+                            lambda x: x['path'] in self.included,
+                            res.json()
+                        )
+                    )
+                self._state = '{:.2f}'.format(
+                    to_unit(
+                        sum([data['freeSpace'] for data in self.data]),
+                        self._unit
+                    )
+                )
+            elif self.type == 'status':
+                self.data = res.json()
+                self._state = self.data['version']
+            self._available = True
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format('Radarr', self._name)
+
+    @property
+    def state(self):
+        """Return sensor state."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return sensor availability."""
+        return self._available
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of the sensor."""
+        return self._unit
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        attributes = {}
+        if self.type == 'upcoming':
+            for movie in self.data:
+                attributes[to_key(movie)] = get_release_date(movie)
+        elif self.type == 'commands':
+            for command in self.data:
+                attributes[command['name']] = command['state']
+        elif self.type == 'diskspace':
+            for data in self.data:
+                attributes[data['path']] = '{:.2f}/{:.2f}{} ({:.2f}%)'.format(
+                    to_unit(data['freeSpace'], self._unit),
+                    to_unit(data['totalSpace'], self._unit),
+                    self._unit, (
+                        to_unit(data['freeSpace'], self._unit) /
+                        to_unit(data['totalSpace'], self._unit) * 100
+                    )
+                )
+        elif self.type == 'movies':
+            for movie in self.data:
+                attributes[to_key(movie)] = movie['downloaded']
```

Comment:
I also changed the attribute value to 'downloaded' instead of 'hasFile'. I honestly don't know which one is preferred.